src/modules/pair/pair.resolver.ts

```diff
@@ -31,6 +32,62 @@ import { constantsConfig } from 'src/config';
 import { GenericResolver } from 'src/services/generics/generic.resolver';
 import { FarmComputeServiceV2 } from '../farm/v2/services/farm.v2.compute.service';
 import { StakingComputeService } from '../staking/services/staking.compute.service';
+import { StakingProxyService } from '../staking-proxy/services/staking.proxy.service';
+import { NftCollection } from '../tokens/models/nftCollection.model';
+import { EnergyService } from '../energy/services/energy.service';
+
+@Resolver(() => PairRewardTokensModel)
+export class PairRewardTokensResolver extends GenericResolver {
+    constructor(
+        private readonly pairCompute: PairComputeService,
+        private readonly pairService: PairService,
+        private readonly stakingProxyService: StakingProxyService,
+        private readonly energyService: EnergyService,
+    ) {
+        super();
+    }
+
+    @ResolveField()
+    async poolRewards(
+        @Parent() parent: PairRewardTokensModel,
+    ): Promise<EsdtToken[]> {
+        return await Promise.all([
+            this.pairService.getFirstToken(parent.address),
+            this.pairService.getSecondToken(parent.address),
+        ]);
+    }
+
+    @ResolveField()
+    async farmReward(
+        @Parent() parent: PairRewardTokensModel,
+    ): Promise<NftCollection> {
+        const farmAddress = await this.pairCompute.getPairFarmAddress(
+            parent.address,
+        );
+
+        if (!farmAddress) {
+            return undefined;
+        }
+
+        return await this.energyService.getLockedToken();
```

Comment:
i don't see any reference of farmAddress of being used besides the undefined check. is it expected? something like 'if the farm address exists, return the general locked token (even tho it has 'nothing' to to with the address)?'